tests/test_s3_functions.py

```diff
@@ -0,0 +1,123 @@
+import boto3
+from moto import mock_aws
+import pytest
+
+from dpytools.s3.basic import (
+    get_s3_object,
+    read_s3_file_content,
+    read_s3_file_content_as_dict,
+    download_s3_file_content_to_local,
+    upload_local_file_to_s3
+)
+
+@pytest.fixture
+@mock_aws
+def mock_s3_client():
+    return boto3.client('s3')
+
+
+@mock_aws
+def test_get_s3_object(mock_s3_client):
+    mock_s3_client.create_bucket(Bucket='mybucket', CreateBucketConfiguration={
+        'LocationConstraint': "eu-west-1"
+    })
+    mock_s3_client.put_object(
+        Bucket='mybucket',
+        Body="myvalue",
+        Key="mykey"
+    )
+    
+    result = get_s3_object('mybucket/mykey')
+    assert result['Body'].read() == b'myvalue'
+
+
+@mock_aws
+def test_read_s3_file_content(mock_s3_client):
+    mock_s3_client.create_bucket(Bucket='mybucket', CreateBucketConfiguration={
+        'LocationConstraint': "eu-west-1"
+    })
+    mock_s3_client.put_object(
+        Bucket='mybucket',
+        Body="myvalue",
+        Key="mykey"
+    )
+    result = read_s3_file_content('mybucket/mykey')
+    assert result == b"myvalue"
+
+
+@mock_aws
+def test_read_s3_file_content_as_dict(mock_s3_client):
+    mock_s3_client.create_bucket(Bucket='mybucket', CreateBucketConfiguration={
+        'LocationConstraint': "eu-west-1"
+    })
+    mock_s3_client.put_object(
+        Bucket='mybucket',
+        Body=b'{"key":"value"}',
+        Key="mykey.json"
+    )
+    result = read_s3_file_content_as_dict('mybucket/mykey.json')
+    assert result == {"key":"value"}
+
+
+@mock_aws
+def test_read_s3_file_content_as_dict_raises_without_json_extension():
+    """
+    Attempting to use read_s3_file_content_as_dict against a non json
+    file extension should raise a value error.
+    """
+    with pytest.raises(ValueError):
```

Comment:
This will work, but could we have an assertion/check for the resulting error message? ("Object name must end with '.json'")